src/actions/backup-environment/index.ts

```diff
@@ -12,12 +12,24 @@ import { AuthHandler, AuthKind, DefaultRunnerFactory, RunnerFactory} from '../..
 export async function main(factory: RunnerFactory): Promise<void> {
     try {
         core.startGroup('backup-environment:');
-        const envUrl = core.getInput('environment-url', { required: true });
+        const envUrl = core.getInput('environment-url', { required: false });
+        const envId = core.getInput('environment-id', { required: false });
         const backupLabel = core.getInput('backup-label', {required: true });
 
         const pac = factory.getRunner('pac', process.cwd());
         await new AuthHandler(pac).authenticate(AuthKind.ADMIN);
-        const backupEnvArgs = ['admin', 'backup', '--url', envUrl, '--label', backupLabel];
+
+        let backupEnvArgs;
+        if (envUrl) {
+            backupEnvArgs = ['admin', 'backup', '--url', envUrl, '--label', backupLabel];
+        } else if (envId) {
+            backupEnvArgs = ['admin', 'backup', '-id', envId, '--label', backupLabel];
+        } else {
+            throw new Error(
+                "Must provide either environment-id or environment-url!"
```

Comment:
We should move this validation logic up a few lines, prior to the creation of the `pac` runner and authentication, to fail sooner.